src/Microsoft.DotNet.Build.Tasks/Targets/publishtest.targets

```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="ResolveNuGetPackages" AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"/>
+
+  <PropertyGroup>
+    <PrepareForRunDependsOn>
+      $(PrepareForRunDependsOn);
+      CopyTestToTestDirectory;
+    </PrepareForRunDependsOn>
+    
+    <CopyTest Condition="'$(CopyTest)'=='' And $([System.String]::new('$(MSBuildProjectName)').EndsWith('.Tests'))">True</CopyTest>
+    <CopyTest Condition="'$(CopyTest)'=='' And !$([System.String]::new('$(MSBuildProjectName)').EndsWith('.Tests'))">False</CopyTest>
+  </PropertyGroup>
+  
+  <ItemGroup>
+    <TestTargetFramework Include="ASP.NetCore, version=v5.0">
+      <Folder>aspnetcore50</Folder>
+    </TestTargetFramework>
+  </ItemGroup>
+
+  <Target Name="CopyTestToTestDirectory"
+    Outputs="$(TestPath)%(TestTargetFramework.Folder)\*.*"
+    Condition="'$(CopyTest)'=='True'">
+    
+    <ItemGroup>
+      <PackagesConfigs Include="$(PackagesConfigFile)"/>
+      <PackagesConfigs Include="$(BaseIntermediateOutputPath)test-runtime\packages.config" Condition="Exists('$(BaseIntermediateOutputPath)test-runtime\packages.config')"/>
+    </ItemGroup>
+    
+    <ResolveNuGetPackages PackagesConfigs="@(PackagesConfigs)"
+                          PackageRoot="$(PackagesDir)"
+                          Platform="$(PlatformTarget)"
+                          Configuration="$(NuGetConfiguration)"
+                          Language="$(Language)"
+                          TargetFramework="%(TestTargetFramework.Identity)"
+                          TargetPlatformMoniker="$(TargetPlatformMoniker)">
+
+      <Output TaskParameter="ResolvedAnalyzers" ItemName="None" />
+      <Output TaskParameter="ResolvedReferences" ItemName="None" />
```

Comment:
If you don't actually need these outputs you can just remove them as opposed to putting them in the None itemgroup.